test/system_settings_test.rb

```diff
@@ -0,0 +1,361 @@
+        .with(Yast::Path.new(".etc.sysctl_conf.\"kernel.sysrq\""))
+        .and_return(sysctl_sysrq)
+      allow(Yast::SCR).to receive(:Read)
+        .with(Yast::Path.new(".target.string"), "/proc/sys/kernel/sysrq")
+        .and_return(kernel_sysrq)
+      allow(Yast::Bootloader).to receive(:kernel_param)
+        .with(:common, "elevator").and_return(scheduler)
+    end
+
+    context "when SysRq keys are enabled via sysctl" do
+      let(:sysctl_sysrq) { "1" }
+
+      it "enables SysRq keys" do
+        settings.Read
+        expect(settings.GetSysRqKeysEnabled).to eq(true)
+      end
+    end
+
+    context "when SysRq keys are disabled via sysctl" do
+      let(:sysctl_sysrq) { "0" }
+
+      it "disables SysRq keys" do
+        settings.Read
+        expect(settings.GetSysRqKeysEnabled).to eq(false)
+      end
+    end
+
+    context "when SysRq keys configuration cannot be read" do
+      let(:kernel_sysrq) { "0" }
+      let(:sysctl_sysrq) { nil }
+
+      it "uses kernel value" do
+        settings.Read
+        expect(settings.GetSysRqKeysEnabled).to eq(false)
+      end
+    end
+
+    context "when scheduler parameter is missing" do
+      let(:scheduler) { :missing }
+
+      it "unsets IO scheduler" do
+        settings.Read
+        expect(settings.GetIOScheduler).to eq("")
+      end
+    end
+
+    context "when scheduler parameter is present but does not have a value" do
+      let(:scheduler) { :present }
+
+      it "unsets IO scheduler" do
+        settings.Read
+        expect(settings.GetIOScheduler).to eq("")
+      end
+    end
+
+    context "when scheduler parameter has a valid value" do
+      it "sets IO scheduler to that value" do
+        settings.Read
+        expect(settings.GetIOScheduler).to eq(scheduler)
+      end
+    end
+
+    context "when scheduler parameter has an invalid value" do
+      let(:scheduler) { "some-scheduler" }
+
+      it "unsets IO scheduler" do
+        settings.Read
+        expect(settings.GetIOScheduler).to eq("")
+      end
+    end
+
+    it "reads bootloader configuration" do
+      expect(Yast::Bootloader).to receive(:Read)
+      settings.Read
+    end
+
+    context "is not in normal mode" do
+      before do
+        allow(Yast::Mode).to receive(:mode).and_return("installation")
+      end
+
+      it "does not read bootloader configuration" do
+        expect(Yast::Bootloader).to_not receive(:Read)
+        settings.Read
+      end
+
+    end
+  end
+
+  describe "#Activate" do
+    let(:sysrq_keys) { false }
+    let(:scheduler)  { "" }
+    let(:disk)       { "/sys/block/sda/queue/scheduler" }
+
+    before do
+      settings.SetSysRqKeysEnabled(sysrq_keys)
+      settings.SetIOScheduler(scheduler)
+      allow(Yast::Bootloader).to receive(:modify_kernel_params)
+      allow(Yast::Bootloader).to receive(:proposed_cfg_changed=)
+      allow(Dir).to receive(:[]).with(/scheduler/).and_return([disk])
+    end
+
+    context "given that SysRq keys status is unknown" do
```

Comment:
NP: I'd remove the "given that ..." beginning, the context is "SysRq keys status is unknown", IMHO it's redundant.